fix(lvs/replica): destroy using pool identifier

tiagolobocastro opened the pull request:
Adding the pool to the request allows us determine whether a pool is loaded or not. Example: if a pool is not loaded then we cannot destroy the replica and we must return an adequate error code back as it's the control-plane job to import pools before attempting to destroy replicas.

tiagolobocastro committed:
fix(lvs/replica): destroy using pool identifier

Adding the pool to the request allows us determine whether a pool is loaded or not.
Example: if a pool is not loaded then we cannot destroy the replica and we must return an
adequate error code back as it's the control-plane job to import pools before attempting
to destroy replicas.

To distinguish from previous not found implementations, a new header with the gtm ticket
number if added.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors try

dsharma-dc reviewed (approved):
nitpick: The commit title mentions "parent pool", which sounds like there is some hierarchical relation between 'pools' rather. Can we just say "destroy replica using pool identifier" ?

mtzaurus commented on io-engine/src/grpc/v1/replica.rs:
```diff
@@ -203,22 +203,57 @@ impl ReplicaRpc for ReplicaService {
+                    })?;
+
+                if let Some(lvs) = lvs {
+                    let same = lvs.name() != lvol.pool_name()
```
I might be overlooking, but isn't the naming a bit confusing here ? Intuitively, 'same' is when something equals, however we check the inequality here. Also, 'same' will be true when name doesn't match the pool name, but in the error message below we inform about equality ('does match').

tiagolobocastro replied:
Yes indeed, amended.

tiagolobocastro committed:
chore: update named_function dependency

This will allow us to use let else within named functions.
Bonus: reduces duplicate dependencies

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

tiagolobocastro commented:
bors merge
